src/models/d2ModelFactory.ts

```diff
@@ -12,5 +48,13 @@ const classes: { [modelName: string]: typeof D2Model } = _.keyBy(metadataModels,
 export function d2ModelFactory(d2: D2, d2ModelName: string): typeof D2Model {
     const modelName = d2.models[d2ModelName].name;
     const className = modelName.charAt(0).toUpperCase() + modelName.slice(1) + "Model";
-    return classes[className] || defaultModel(modelName);
+
+    let model = classes[className];
```

Comment:
We use to use ``const`` for almost everything. I believe we could here keep the ``return classes[className] || defaultModel(modelName);`` and just add before the return an ``if (!classes[class ame])``.